dspace-api/src/main/java/org/dspace/subscriptions/dSpaceObjectsUpdates/ItemUpdates.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.subscriptions.dSpaceObjectsUpdates;
+
+import static org.apache.commons.lang3.StringUtils.trimToEmpty;
+
+import java.sql.SQLException;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.CommunityService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.discovery.DiscoverQuery;
+import org.dspace.discovery.DiscoverResult;
+import org.dspace.discovery.IndexableObject;
+import org.dspace.discovery.SearchService;
+import org.dspace.discovery.configuration.DiscoveryConfiguration;
+import org.dspace.discovery.configuration.DiscoveryConfigurationService;
+import org.dspace.discovery.configuration.DiscoveryRelatedItemConfiguration;
+import org.dspace.discovery.configuration.DiscoverySortConfiguration;
+import org.dspace.discovery.configuration.DiscoverySortFieldConfiguration;
+import org.dspace.discovery.configuration.DiscoverySortFunctionConfiguration;
+import org.dspace.discovery.indexobject.IndexableCollection;
+import org.dspace.discovery.indexobject.IndexableCommunity;
+import org.dspace.discovery.indexobject.IndexableItem;
+import org.dspace.subscriptions.service.DSpaceObjectUpdates;
+
+/**
+ * Class which will be used to find
+ * all item objects updated related with subscribed DSO
+ *
+ * @author Alba Aliu
+ */
+public class ItemUpdates implements DSpaceObjectUpdates {
```

Comment:
I suggest to get ride of this class and all the related code. This will make the PR smaller and easier to finalize.
The goal of this class was to retrieve a list of updates related to a specific item (i.e. a Person Item), this would be a new feature compared to what exists in previous version of DSpace so it would be ok for this PR to just focus on Communities and Collections (the only ones supported in previous versions) demonstrating that it could be further extended in future to support other scenario (via followup PRs)

Reply:
@abollini & @corrad82-4s : It's reasonable to me to remove the `ItemUpdates` and the ability to subscribe to an Item.  It is not a requirement. The only requirement is that we must be able to subscribe to a *Collection* (as that was all that was possible in 6.x).  Keep in mind, if we remove `ItemUpdates`, then we need to ensure this item subscription feature is fully removed from both the frontend & backend code.